src/sentry/static/sentry/app/views/organizationGroupDetails/reprocessingDialogForm.tsx

```diff
@@ -53,46 +59,69 @@ function ReprocessingDialogForm({group, organization, Header, Body, closeModal}:
-            <ListItem key={index}>{impact}</ListItem>
-          ))}
-        </StyledList>
-        <ApiForm
-          apiEndpoint={endpoint}
-          apiMethod="POST"
-          footerClass="modal-footer"
-          onSubmitSuccess={handleSuccess}
-          submitLabel={title}
-          submitLoadingMessage={t('Reprocessing\u2026')}
-          submitErrorMessage={t('Failed to reprocess. Please check your input.')}
-          onCancel={closeModal}
-        >
-          <NumberField
-            name="maxEvents"
-            label={t('Enter the number of events to be reprocessed')}
-            help={tct(
-              'You can limit the number of events reprocessed in this Issue. If you set a limit, we will reprocess your most recent events, [strong:and the rest will be deleted.]',
-              {strong: <strong />}
+  render() {
+    const {group, organization, Header, Body, closeModal} = this.props;
+    const {maxEvents} = this.state;
+    const orgSlug = organization.slug;
+    const endpoint = `/organizations/${orgSlug}/issues/${group.id}/reprocessing/`;
+    const title = t('Reprocess Events');
+
+    return (
+      <React.Fragment>
+        <Header closeButton>{title}</Header>
+        <Body>
+          <Introduction>
+            {t(
+              'Reprocessing applies any new debug files or grouping configuration to an Issue. Before you give it a try, you should probably consider these impacts:'
             )}
-            placeholder={t('Reprocess all events')}
-            min={1}
-          />
-        </ApiForm>
-      </Body>
-    </React.Fragment>
-  );
+          </Introduction>
+          <StyledList symbol="bullet">
+            {impacts.map((impact, index) => (
+              <ListItem key={index}>{impact}</ListItem>
+            ))}
+          </StyledList>
+          <Form
+            submitLabel={title}
+            apiEndpoint={endpoint}
+            apiMethod="POST"
+            initialData={{maxEvents: '', remainingEvents: 'keep'}}
+            onSubmitSuccess={this.handleSuccess}
+            onSubmitError={this.handleError}
+            onCancel={closeModal}
+            footerClass="modal-footer"
+          >
+            <NumberField
+              name="maxEvents"
+              label={t('Number of events to be reprocessed')}
+              help={t('If you set a limit, we will reprocess your most recent events.')}
+              placeholder={t('Reprocess all events')}
+              value={maxEvents}
+              onChange={v => this.setState({maxEvents: v.toString()})}
+              min={1}
+            />
+
+            <RadioField
+              orientInline
+              label={t('Remaining events')}
+              help={t('What to do with the events that are not reprocessed.')}
+              name="remainingEvents"
+              choices={[
+                ['keep', t('Keep')],
+                ['delete', t('Delete')],
+              ]}
+              disabled={maxEvents === ''}
```

Comment:
I would add here `maxEvents.trim() `but as the `maxEvent` field is a number, it should be fine